posthog/temporal/workflows/batch_exports.py

```diff
@@ -634,6 +635,14 @@ async def execute_batch_export_insert_activity(
         initial_retry_interval_seconds: When retrying, seconds until the first retry.
         maximum_retry_interval_seconds: Maximum interval in seconds between retries.
     """
+
+    def capture_exception_with_metadata(e: BaseException):
+        with push_scope() as scope:
+            scope.set_extra("run_id", update_inputs.id)
+            scope.set_tag("team_id", update_inputs.team_id)
+            scope.set_tag("workflow_type", workflow.info().workflow_type.lower())
+            capture_exception(e)
```

Comment:
Are we worried about this blocking? I'm not too familiar with the sentry sdk.

Reply:
Unfortunately, this is in the Workflow event loop, so we cannot do `asyncio.to_thread`.

Reply:
Oh interesting, I know you mentioned the threading thing previously. Do they hard prevent us from spawning threads?

Sentry SDK does its HTTP requests in a threaded Worker: https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/worker.py#L21

Reply:
Hmm, so I understand [Workflows should be deterministic](https://docs.temporal.io/dev-guide/python/foundations#workflow-logic-requirements), but sending an event to Sentry shouldn't really mess with our Workflow itself, and the worst case scenario is we double-log to Sentry on a retry, right?

So it's not clear to me whether we *can't* use threads (i.e. `asyncio.to_thread`, however Sentry uses threads itself), or that we're not supposed to?

Reply:
`asyncio.to_thread` we straight up can't because Temporal runs workflows on their own event loop that doesn't implement any thread-related methods.

Reply:
There shouldn't be any problems with spawning the threads ourselves though

Reply:
I think optimally we want our own Sentry interceptor: https://github.com/temporalio/samples-python/blob/68ba233c5c8e1073d4f5a5c5db9e6e86c940a332/sentry/interceptor.py#L71